packages/pipewire/build.sh

```diff
@@ -0,0 +1,38 @@
+TERMUX_PKG_HOMEPAGE=https://pipewire.org/
+TERMUX_PKG_DESCRIPTION="A server and user space API to deal with multimedia pipelines."
+TERMUX_PKG_LICENSE="MIT, LGPL-2.1, LGPL-3.0, GPL-2.0"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=0.3.51
+TERMUX_PKG_SRCURL="https://gitlab.freedesktop.org/pipewire/pipewire/-/archive/${TERMUX_PKG_VERSION}/pipewire-${TERMUX_PKG_VERSION}.tar.bz2"
+TERMUX_PKG_SHA256=1820bfd524805ef3e4e4dcb368f26efa5f43ca90f8c3c67731e502268c34245e
+TERMUX_PKG_DEPENDS="avahi, ffmpeg, libfdk-aac, liblua54, lilv, libsndfile, libusb, libwebrtc-audio-processing, pulseaudio"
+
+# 'media-session' session-managers is disabled as it requires alsa.
+# Since we are building without x11, dbus is disabled.
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
+-Dtests=disabled
+-Dexamples=disabled
+-Dpipewire-alsa=disabled
+-Dalsa=disabled
+-Dpipewire-jack=disabled
+-Djack=disabled
+-Ddbus=disabled
+-Dsession-managers=['wireplumber']
+-Dffmpeg=enabled
+-Dwireplumber:system-lua=true
+-Dwireplumber:system-lua-version=54
+"
+
+termux_step_pre_configure() {
+	CFLAGS+=" -Dindex=strchr -Drindex=strrchr"
+	sed "s|@TERMUX_PKG_BUILDER_DIR@|${TERMUX_PKG_BUILDER_DIR}|g" \
+		"${TERMUX_PKG_BUILDER_DIR}"/pipewire.meson.build.diff | patch -p1
+	sed "s|@TERMUX_PKG_BUILDER_DIR@|${TERMUX_PKG_BUILDER_DIR}|g" \
+		"${TERMUX_PKG_BUILDER_DIR}"/modules.meson.build.diff | patch -p1
```

Comment:
Might be easier to copy reallocarray.c to the src directory, and convert the .diff's to .patch

Reply:
Then we would have to copy to multiple places in source-tree (almost same amount of work). Though I have merged both diff into one.

Also, this can last longer without edits as, if ever dirs in source-tree is changed we would only need to update diff.